WeatherForecast/WeatherForecast/ViewController.swift

```diff
@@ -5,14 +5,164 @@
+        if let fiveDayForercast = fiveDayForecast {
+            cell.setUp(with: fiveDayForercast, of: indexPath)
+        }
+        cell.selectionStyle = .none
+        
+        return cell
+    }
+}
 
+extension ViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        guard let view = weatherTableView.dequeueReusableHeaderFooterView(withIdentifier: CurrentWeatherHeader.headerIdentifier) as? CurrentWeatherHeader else {
+            return UIView()
+        }
+        if let currentWeather = currentWeather {
+            view.setUp(with: currentWeather, address)
+        }
+        
+        return view
+    }
+}
+
+extension ViewController: CLLocationManagerDelegate {
+    func locationManager(_ manager: CLLocationManager, didUpdateLocations locations: [CLLocation]) {
+        guard let latitude = locations.first?.coordinate.latitude,
+              let longitude = locations.first?.coordinate.longitude else { return }
+        fetchCurrentWeather(latitude: latitude, longitude: longitude)
+        fetchFiveDayForecast(latitude: latitude, longitude: longitude)
+        
+        let location = CLLocation(latitude: latitude, longitude: longitude)
+        let geoCoder = CLGeocoder()
+        let locale = Locale(identifier: "Ko-kr")
+        geoCoder.reverseGeocodeLocation(location, preferredLocale: locale) { placemarks, error in
+            self.address = placemarks
+        }
+        
+        locationManager.stopUpdatingLocation()
+    }
 }
 
+extension ViewController {
+    private func setUpTableView() {
+        view.addSubview(weatherTableView)
+        weatherTableView.sectionFooterHeight = .zero
+        weatherTableView.backgroundColor = UIColor.clear
+        weatherTableView.dataSource = self
+        weatherTableView.delegate = self
+        weatherTableView.register(ThreeHourForecastCell.self, forCellReuseIdentifier: ThreeHourForecastCell.cellIdentifier)
+        weatherTableView.register(CurrentWeatherHeader.self,
+                                  forHeaderFooterViewReuseIdentifier: CurrentWeatherHeader.headerIdentifier)
+    }
+    
+    private func setUpLayouts() {
+        weatherTableView.translatesAutoresizingMaskIntoConstraints = false
+        
+        NSLayoutConstraint.activate([
+            weatherTableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor),
+            weatherTableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor),
+            weatherTableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+            weatherTableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor)
+        ])
+    }
+    
+    private func setBackgroundImage() {
+        let imageView = UIImageView()
+        imageView.image = UIImage(named: "weatherAppBg")
+        imageView.contentMode = .scaleAspectFill
+        view.addSubview(imageView)
```

Comment:
background image 의 경우 가장 첫번째 (맨뒤) subview로 add되어야 하기 때문에 
0번째 인덱스에 insert 해주는 건 어떨까요? 
```suggestion
        view.insertSubview(imageView, at: 0)
```

Reply:
`insertSubview()`라는 메서드가 있군요..! 찾아보았는데 `addSubview`보다 훨씬 백그라운드이미지 설정에 적합한 메서드인 것 같습니다. 감사합니다!